### PR TITLE
Support int and uint as datatype for template formatting

### PR DESCRIPTION
**Pull request by jebeaudet**
Fixes GH-9679

### CODE REVIEW

**Comment by roidelapluie** [comment]
Converted as a draft until I think about https://github.com/prometheus/prometheus/issues/9679#issuecomment-962161263

File: template/template.go [hunk 1]
```diff
@@ -286,7 +291,11 @@ func NewTemplateExpander(
 				if math.IsNaN(v) || math.IsInf(v, 0) {
 					return fmt.Sprintf("%.4g", v), nil
 				}
-				t := model.TimeFromUnixNano(int64(v * 1e9)).Time().UTC()
+				timestamp := v * 1e9
+				if timestamp > math.MaxInt64 || timestamp < math.MinInt64 {
+					return "", fmt.Errorf("%T cannot be represented as a nanoseconds timestamp since it overflows int64", v)
```
Comment (roidelapluie) [inline template/template.go:296]: I'd rather print the value here maybe, because %T can only be float

**Comment by roidelapluie** [comment]
Thanks for your PR. For legal reasons, we require that all commits are signed with a [DCO](https://developercertificate.org/) before we can merge them. See [this blog post](https://www.cncf.io/blog/2016/11/08/cloud-native-software-can-trust/) for considerations around this.

This means that the last line of your commit message should read like:

`Signed-Off-By: Your Name <your@email.address>`

If you are using GitHub through the web interface, it's quickest to close this PR and open a new one with the appropriate line.

If you are using Git on the command line, it is probably quickest to amend and force push. You can do that with
````
git commit --amend --signoff
git push -f $remote $remote_branch_for_pr
````

As always, be careful when force-pushing.

File: template/template.go (hunk 1 above)
Comment (jebeaudet) [reply template/template.go:296]: yup good catch, fixing

**Comment by jebeaudet** [comment]
@roidelapluie Done for the DCO, mismatched my professional email for my personal email in the signed off thing, that's why it didn't match before.

**Review by roidelapluie — APPROVED** [review]

**Comment by roidelapluie** [comment]
thanks!

**Pull request merged**